tests/integration/url_test.go

```diff
@@ -0,0 +1,87 @@
+package integration
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odoURLIntegration", func() {
+	//new clean project and context for each test
+	var project string
+	var context string
+
+	//  current directory and project (before eny test is run) so it can restored  after all testing is done
+	var originalDir string
+	var originalProject string
+	var oc helper.OcRunner
+
+	// Setup up state for each test spec
+	// create new project (not set as active) and new context directory for each test spec
+	// This is before every spec (It)
+	var _ = BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		oc = helper.NewOcRunner("oc")
+		project = helper.CreateRandProject()
+		context = helper.CreateNewContext()
+		originalDir = helper.Getwd()
+		originalProject = oc.GetCurrentProject()
+		helper.Chdir(context)
+		oc.SwitchProject(project)
+	})
+
+	// Clean up after the test
+	// This is run after every Spec (It)
+	var _ = AfterEach(func() {
+		helper.Chdir(originalDir)
+		oc.SwitchProject(originalProject)
+		helper.DeleteProject(project)
+		helper.DeleteDir(context)
+	})
+
+	Context("Listing urls", func() {
```

Comment:
Test spec looks too length due to lots of text validation. Can you please minimize those text assertion by keeping those text which are really matter for the spec and output. And more importantly keep all the url command scenario here those are present in other test file like ```generic_test.go```.

If possible add all possible url command and sub-command in this test file.

Reply:
Sure, I will see if i can shorten it.

Reply:
Yes, i guess it would be possible to do that, but `generic _test.go` contains a lot of tests that may as well reside in a specific file. I think it is ok to have that there and have URL specific tests here.

Reply:
> I think it is ok to have that there and have URL specific tests here.

You are right, I mean to say just check the UX if anything is missed in the URL specific test file. 

For example:
```
$ odo url -h
Expose component to the outside world. 

The URLs that are generated using this command, can be used to access the deployed components from outside the cluster.

Usage:
  odo url [flags]
  odo url [command]

Examples:
  # Create a URL for the current component with a specific port
  odo url create --port 8080
  
  # Create a URL with a specific name and port
  odo url create example --port 8080
  
  # Create a URL with a specific name by automatic detection of port (only for components which expose only one service port)
  odo url create example
  
  # Create a URL with a specific name and port for component frontend
  odo url create example --port 8080 --component frontend
  # Delete a URL to a component
  odo url delete myurl
  # List the available URLs
  odo url list

Available Commands:
  create      Create a URL for a component
  delete      Delete a URL
  list        List URLs

Flags:
  -h, --help   Help for url

Global Flags:
      --skip-connection-check   Skip cluster check
  -v, --v Level                 Log level for V logs. Level varies from 0 to 9 (default 0).
      --vmodule moduleSpec      Comma-separated list of pattern=N settings for file-filtered logging

Use "odo url [command] --help" for more information about a command.
```